ReactAndroid/src/main/java/com/facebook/react/views/webview/ReactWebViewManager.java

```diff
@@ -317,7 +317,11 @@ public void setSource(WebView view, @Nullable ReadableMap source) {
           ReadableMapKeySetIterator iter = headers.keySetIterator();
           while (iter.hasNextKey()) {
             String key = iter.nextKey();
-            headerMap.put(key, headers.getString(key));
+            if ("user-agent".equals(key.toLowerCase(Locale.ENGLISH))) {
+              view.getSettings().setUserAgentString(headers.getString(key));
```

Comment:
I just discovered that you've already included this as it's own prop. I'll use this instead of having this redundancy in the source prop.